### PR TITLE
Unify links

### PR DESCRIPTION
**Pull request by brichet**
This PR unifies the links by creating a unique schema/interface for advanced links and component links.

These links are loaded in the shared glue session to use it the same way in the frontend and in the backend (in *glue_session.py* in a follow up PR).

It also fixes the special case of the advanced links in the *General* category, which are actually component links.

### CODE REVIEW

File: src/linkPanel/widgets/linking.ts [hunk 1]
```diff
@@ -445,17 +526,31 @@ namespace Private {
...
-        select.value = attributes[index] || attributes[0];
-        this._io[ioType].push(attributes[index] || attributes[0]);
-
         select.onchange = () => {
-          this.onSelectIO(ioType, index, select.value);
+          this.onSelectIO(
+            ioType,
+            index,
+            select.value,
+            (select.querySelector('option:selected') as HTMLOptionElement).label
```
Comment (trungleduc) [inline src/linkPanel/widgets/linking.ts:539]: The `onchange` callback will receive a change event as its parameter where you can get the value and label of the selected item.

File: glue_lab/glue_ydoc.py [hunk 2]
```diff
@@ -128,13 +128,45 @@ def set(self, value: str) -> None:
 
         links: Dict[str, Dict] = {}
         for link_name in link_names:
-            links[link_name] = contents.get(link_name, {})
-            if links[link_name]["_type"] != COMPONENT_LINK_TYPE:
-                for i in range(1, 3):
-                    listName = links[link_name][f"cids{i}"]
-                    links[link_name][f"cids{i}"] = contents.get(listName, {}).get(
+            link: Dict = contents.get(link_name, {})
```
Comment (trungleduc) [inline glue_lab/glue_ydoc.py:131]: Could you add a test like this one?  https://github.com/QuantStack/glue-lab/blob/be0e40a503ae96c01710642e65819c2716310d38/glue_lab/tests/test_ydoc.py#L15-L22
for a session having link data on it.

**Comment by brichet** [comment]
Thanks @trungleduc, I added a test on the links in the ydocument and simplified the advanced attribute selection.

**Comment by trungleduc** [comment]
Thanks @brichet

**Review by trungleduc — APPROVED** [review]

**Pull request merged**